multimod/internal/common/git.go

```diff
@@ -155,3 +156,53 @@ func VerifyWorkingTreeClean(repo *git.Repository) error {
 
 	return nil
 }
+
+type Client interface {
+	HeadCommit(r *git.Repository) (*object.Commit, error)
+	TagCommit(r *git.Repository, tag string) (*object.Commit, error)
+	FilesChanged(headCommit *object.Commit, tagCommit *object.Commit, prefix string, suffix string) ([]string, error)
+}
+
+type GitClient struct{}
+
+func (g GitClient) HeadCommit(r *git.Repository) (*object.Commit, error) {
+	headRef, err := r.Head()
+	if err != nil {
+		return nil, err
+	}
+	return r.CommitObject(headRef.Hash())
+}
+
+func (g GitClient) TagCommit(r *git.Repository, tag string) (*object.Commit, error) {
+	tagRef, err := r.Tag(tag)
+	if err != nil {
+		return nil, err
+	}
+
+	o, err := r.TagObject(tagRef.Hash())
+	if err != nil {
+		return nil, fmt.Errorf("tag object error %s %w", tagRef.Hash().String(), err)
+	}
+	return r.CommitObject(o.Target)
+}
+
+// FilesChanged returns the differences between two commits
```

Comment:
This comment is inaccurate. It only list files. It does not really say what are the differences.